Content.Server/GameTicking/Rules/TraitorRuleSystem.cs

```diff
@@ -129,20 +139,50 @@ public bool MakeTraitor(EntityUid traitor, TraitorRuleComponent component, bool
         return true;
     }
 
+    private (Note[]?, string) RequestUplink(EntityUid traitor, FixedPoint2 startingBalance, string briefing)
+    {
+        var pda = _uplink.FindUplinkTarget(traitor);
+        Note[]? code = null;
+
+        var uplinked = _uplink.AddUplink(traitor, startingBalance, giveDiscounts: true);
```

Comment:
```suggestion
        var uplinked = _uplink.AddUplink(traitor, startingBalance, pda, true);
```
pass the pda into AddUplink so that the function does not have to call FindUplinkTarget again